examples/sample_app.py

```diff
@@ -0,0 +1,39 @@
+from snowfakery import generate_data, ParentApplication
+
+
+class MyApplication(ParentApplication):
+    def __init__(self, stopping_tablename, count):
+        self.count = count
+        self._stopping_tablename
+
+    def echo(self, message=None, file=None, nl=True, err=False, color=None):
+        """Write something to a virtual stdout or stderr.
+
+        Arguments to this function are derived from click.echo"""
+        # this is a simplistic implementation! Model your real one
+        # on click.echo
+        if err:
+            print("ERROR:" + message)
+        else:
+            print(message)
+
+    @property
+    def stopping_tablename(self):
+        """Return the name of "stopping table/object":
+
+        The table/object whose presence determines
+        whether we are done generating data.
+
+        This is used by Snowfakery to validate that
+        the provided recipe will not generate forever
+        due to a misspelling the stopping tablename."""
+        return self._stopping_tablename
+
+    def check_if_finished(self, id_manager):
+        """Check if we're finished generating"""
+        print(id_manager[self.stopping_tablename])
+        return id_manager[self.stopping_tablename] >= self.count
+
+
+myapp = MyApplication("Employee", 100)
+generate_data("examples/company.yml", parent_application=myapp, output_file="out.json")
```

Comment:
Is there a way to include this example from embedding.md to avoid duplication / potential for getting out of sync?

Reply:
Yes thanks for reminding me that I have such a tool. When I figure out the pre-commit hooks it will update automatically.